File: api/src/main/java/org/pmiops/workbench/audit/adapters/WorkspaceAuditAdapterService.java
package org.pmiops.workbench.audit.adapters;

import java.util.Map;
import org.pmiops.workbench.db.model.DbWorkspace;
import org.pmiops.workbench.model.Workspace;

public interface WorkspaceAuditAdapterService {
  void fireCreateAction(Workspace createdWorkspace, long dbWorkspaceId);

  void fireDeleteAction(DbWorkspace dbWorkspace);

  void fireDuplicateAction(
      DbWorkspace sourceWorkspaceDbModel,
      DbWorkspace destinationWorkspaceDbModel);

  void fireCollaborateAction(long sourceWorkspaceId, Map<Long, String> aclStringsByUserId);
}

File: api/src/main/java/org/pmiops/workbench/audit/adapters/WorkspaceAuditAdapterService.kt
package org.pmiops.workbench.audit.adapters

import org.pmiops.workbench.model.Workspace

interface WorkspaceAuditAdapterService {
    fun fireCreateAction(createdWorkspace: Workspace, dbWorkspaceId: Long)

    fun fireDeleteAction(dbWorkspace: org.pmiops.workbench.db.model.Workspace)

    fun fireDuplicateAction(
        sourceWorkspaceDbModel: org.pmiops.workbench.db.model.Workspace,
        destinationWorkspaceDbModel: org.pmiops.workbench.db.model.Workspace
    )

    fun fireCollaborateAction(sourceWorkspaceId: Long, aclStringsByUserId: Map<Long, String>)
}
